internal/convert.go

```diff
@@ -168,27 +168,52 @@ func (conv *Conv) SetDataMode() {
 	conv.mode = dataOnly
 }
 
+func dequeue(queue []string) []string {
+	return queue[1:]
+}
+
 // GetDDL Schema returns the Spanner schema that has been constructed so far.
-// Return DDL in alphabetical table order.
+// We sort DDL in alphabetical order, but we maintain the order of interleaved
+// relation by passing tables through a queue
 func (conv *Conv) GetDDL(c ddl.Config) []string {
 	var tables []string
+	var tablesFk []string
 	for t := range conv.SpSchema {
 		tables = append(tables, t)
+		tablesFk = append(tablesFk, t)
```

Comment:
There is no need to compute tables and tablesFk twice.
Suggest just computing tables and then taking a copy i.e. at end of loop:

tablesFk := tables

Actually, suggest switching the roles and keeping tables as the actual set of tables and doing:

tableQueue := tables

and then using tableQueue in the for loop, and then using tables in the loop to print the foreign key contraints.

How about

Reply:
updated function, using tableQueue in first iteration and tables in second.